druid/src/widget/widget_ext.rs

```diff
@@ -159,6 +161,19 @@ pub trait WidgetExt<T: Data>: Widget<T> + Sized + 'static {
         ControllerHost::new(self, controller)
     }
 
+    /// Control the [`LifeCycle::WidgetAdded`] event of this widget with an [`Added`] widget.
+    /// The function provided will be called when the widget is added to the
+    /// widget tree.
+    ///
+    /// [`Added`]: crate::widget::Added
+    /// [`LifeCycle::WidgetAdded`]: crate::LifeCycle::WidgetAdded
```

Comment:
I'm going to nit out here; I think our goal with the docs is to describe what this does and how to use it, and the particular details of the implementation are not important; in particular the user really never needs to know about the `Added` widget. I would go with,

```suggestion
    /// Provide a closure that will be called when this widget is added to the widget tree.
    ///
    /// You can use this to perform any initial setup.
    ///
    /// This is equivalent to handling the [`LifeCycle::WidgetAdded`] event in a
    /// custom [`Controller`].
    ///
    /// [`LifeCycle::WidgetAdded`]: crate::LifeCycle::WidgetAdded
```